pytest/tests/sanity/routing_table_sync.py

```diff
@@ -0,0 +1,185 @@
+
+for (left, right, common, TIMEOUT) in tests:
+    test_id += 1
+    start = time.time()
+    logger.info("test with params left: %s right: %s common: %s" % (left, right, common))
+    time.sleep(1)
+    # set options
+
+    common_nodes = [new_edge(int(2e7), x + 1 + left + right, 2 * test_id + 1) for x in range(common)]
+    left_nodes = [new_edge(0, x + 1, 2 * test_id + 1) for x in range(left)]
+    right_nodes = [new_edge(int(1e7), x + 1 + left, 2 * test_id + 1) for x in range(right)]
+    to_node0 = left_nodes + common_nodes
+    to_node1 = right_nodes + common_nodes
+
+    def simplify(edges):
+        return [{'key': (edge['peer0'], edge['peer1']), 'nonce': edge['nonce']} for edge in edges]
+    all_edges = simplify(left_nodes + to_node1)
+
+    # add edges
+    logger.info("sending new edges")
+
+    for chunk in chunks(to_node0, 10000):
+        nodes[0].json_rpc("adv_set_routing_table", {"add_edges": chunk}, timeout=30)
+    for chunk in chunks(to_node1, 10000):
+        nodes[1].json_rpc("adv_set_routing_table", {"add_edges": chunk}, timeout=30)
+
+    time.sleep(1)
+
+
+    def get_routing_tables(n):
+        logger.info("getting routing tables")
+        var_a = n[0].json_rpc("adv_get_routing_table", {}, timeout=60)["result"]["edges_info"]
+        var_b = n[1].json_rpc("adv_get_routing_table", {}, timeout=60)["result"]["edges_info"]
+        var_a = {(x['key'][0], x['key'][1], x['nonce']) for x in var_a}
+        var_b = {(x['key'][0], x['key'][1], x['nonce']) for x in var_b}
+        return var_a, var_b
+
+
+    # compute set difference of routing tables
+    a, b = get_routing_tables(nodes)
+    logger.info("case 1 nodes %s vs %s diff %s " % (len(a), len(b), len(a.symmetric_difference(b))))
+    assert(len(a) == 1 + left + common)
+    assert(len(b) == 1 + right + common)
+    assert(len(a.symmetric_difference(b)) == left + right)
+
+    time.sleep(1)
+
+    success.value = 0
+    logger.info("starting_sync to %s" % (nodes[0].json_rpc("adv_start_routing_table_syncv2", {
+        "peer_id": peer_id1
+    })))
+
+    # wait for sync
+    started = time.time()
+    while True:
+        assert time.time() - started < TIMEOUT
+        status = nodes[0].get_status()
+        height = status['sync_info']['latest_block_height']
+        if success.value == 1:
+            break
+        time.sleep(1)
+
+    # wait some time for node to process adding edges
+    time.sleep(max(5, TIMEOUT))
+
+    # force synchronization
+
+    a, b = get_routing_tables(nodes)
+    logger.info("case 2 nodes %s vs %s diff %s " % (len(a), len(b), len(a.symmetric_difference(b))))
+    assert(len(a.symmetric_difference(b)) == 0)
+
+    # remove edges
+    logger.info("removing edges")
+    #for chunk in chunks(all_edges, 10000):
+    #     nodes[0].json_rpc("adv_set_routing_table", {"remove_edges": chunk}, timeout=30)
+    #for chunk in chunks(all_edges, 10000):
+    #    nodes[1].json_rpc("adv_set_routing_table", {"remove_edges": chunk}, timeout=30)
```

Comment:
What happened here

Reply:
I accidentally pushed partially changed test. I was in the middle of adding a new check, that tests whenever RemoveEdges works properly or not.
I'll finish that tomorrow.

Reply:
WIP

Reply:
@bowenwang1996 My test is not working, because of existing bug in our code. I'll create a new issue.

Reply:
@matklad I want o write a test that tests edge pruning. Is there a good way to mock time in Rust?
Without having to wait 2 hours for `pub const SAVE_PEERS_MAX_TIME: u64 = 7_200;`.
Of course I can make this into a variable, and set it, but I was wondering if there is a cleaner way.

Reply:
Or I would have to replace all instances of `chrono::Utc::now()` for testing purposes.

Reply:
I changed the test to remove edges using edge pruning. That we will cover RemoveEdges case.